crates/store/re_dataframe/src/latest_at.rs

```diff
@@ -0,0 +1,233 @@
+                            &query,
+                            &descr.entity_path,
+                            [descr.component_name],
+                        );
+
+                        results
+                            .components
+                            .get(&descr.component_name)
+                            .cloned()
+                            .map(|chunk| (descr, chunk))
+                    }
+
+                    _ => None,
+                })
+                .collect()
+        };
+
+        let mut max_time_per_timeline = HashMap::<Timeline, (TimeInt, UnitChunkShared)>::default();
+        {
+            re_tracing::profile_scope!("compound times");
+
+            let timelines = columns
+                .iter()
+                .filter_map(|descr| match descr {
+                    ColumnDescriptor::Time(descr) => Some(descr.timeline),
+                    _ => None,
+                })
+                .collect_vec();
+
+            for unit in all_units.values() {
+                for &timeline in &timelines {
+                    if let Some((time, _)) = unit.index(&timeline) {
+                        max_time_per_timeline
+                            .entry(timeline)
+                            .and_modify(|(cur_time, cur_unit)| {
+                                if *cur_time < time {
+                                    *cur_time = time;
+                                    *cur_unit = unit.clone();
+                                }
+                            })
+                            .or_insert_with(|| (time, unit.clone()));
+                    }
+                }
+            }
+        }
+
+        // NOTE: Keep in mind this must match the ordering specified by `Self::schema`.
+        let packed_arrays = {
+            re_tracing::profile_scope!("packing");
+
+            columns
+                .iter()
+                .filter_map(|descr| match descr {
+                    ColumnDescriptor::Control(_) => {
+                        if cfg!(debug_assertions) {
+                            unreachable!("filtered out during schema computation");
+                        } else {
+                            // NOTE: Technically cannot ever happen, but I'd rather that than an uwnrap.
+                            None
+                        }
+                    }
+
+                    ColumnDescriptor::Time(descr) => {
+                        let time_column = max_time_per_timeline
+                            .remove(&descr.timeline)
+                            .and_then(|(_, chunk)| chunk.timelines().get(&descr.timeline).cloned());
+
+                        Some(if cfg!(debug_assertions) {
+                            #[allow(clippy::unwrap_used)]
+                            time_column.unwrap().times_array().to_boxed()
+                        } else {
+                            // NOTE: Technically cannot ever happen, but I'd rather that than an uwnrap.
```

Comment:
I think this can happen if you request a TimeColumn that was never logged?

My read of the above code is it won't end up with an entry in `max_time_per_timeline`, and as such `null_array` is the right thing to return here.

Reply:
Ha yes, that dates from the time where I would filter out "impossible" columns from `user_columns`.